### PR TITLE
refactor(image-plugin): some improvements

### PR DESCRIPTION
**Pull request by elbotho**
(still works with e2e PR)

### CODE REVIEW

File: src/serlo-editor/plugins/image/controls/settings-modal-controls.tsx [hunk 1]
```diff
@@ -1,15 +1,10 @@
...
 import { ImageProps } from '..'
-import { FaIcon } from '@/components/fa-icon'
 import { useEditorStrings } from '@/contexts/logged-in-data-context'
 import { tw } from '@/helper/tw'
-import { EditorTooltip } from '@/serlo-editor/editor-ui/editor-tooltip'
 import { isTempFile } from '@/serlo-editor/plugin'
 import { OverlayInput } from '@/serlo-editor/plugin/plugin-toolbar'
 
-export function SettingsModalControls({ state, config }: ImageProps) {
+export function SettingsModalControls({ state }: ImageProps) {
```
Comment (hejtful) [inline src/serlo-editor/plugins/image/controls/settings-modal-controls.tsx]:
```suggestion
export function SettingsModalControls({ state }: Pick<ImageProps, 'state'>) {
```

Minor suggestion, makes the type a bit stricter.

File: src/serlo-editor/plugins/image/editor.tsx [hunk 2]
```diff
@@ -25,25 +25,29 @@ export function ImageEditor(props: ImageProps) {
...
-
   useEffect(() => {
     if (editable && !state.caption.defined) {
       state.caption.create({ plugin: EditorPluginType.Text })
     }
   }, [editable, state.caption])
 
+  useEffect(() => {
+    setShowInlineImageUrl(!state.src.value)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (hejtful) [inline src/serlo-editor/plugins/image/editor.tsx:49]: Could you please explain this eslint-disable in a comment?

File: src/serlo-editor/plugins/image/editor.tsx [hunk 3]
```diff
@@ -92,17 +103,4 @@ export function ImageEditor(props: ImageProps) {
       } as TextEditorConfig,
     })
   }
-
-  function renderEditControls() {
```
Comment (hejtful) [inline src/serlo-editor/plugins/image/editor.tsx:96]: I'm a bit confused, I thought you preferred this `render*` approach, but in this plugin toolbar PR, you've been removing a lot of them 😄 What's your preference now? I'd like to align my coding style with yours where possible

**Review by hejtful — APPROVED** [review]

File: src/serlo-editor/plugins/image/editor.tsx (hunk 3 above)
Comment (elbotho) [reply src/serlo-editor/plugins/image/editor.tsx:96]: (see slack)

**Pull request merged**